src/main/java/games/strategy/triplea/attachments/UnitSupportAttachment.java

```diff
@@ -392,7 +392,11 @@ private void addUnitTypes(final Set<UnitType> types) {
     if (m_unitType == null) {
       m_unitType = new HashSet<>();
     }
-    m_unitType.addAll(types);
+    synchronized (m_unitType) {
+      synchronized (types) {
+        m_unitType.addAll(types);
+      }
+    }
```

Comment:
I know :D
Goal was just to temporarily fix the concurrent modificationexceptions, error prone is raising warnings because m_unitType is not final.
Will fix this using a lock later.